### ISSUE TITLE
Backups/Restores Stalling

### ISSUE THREAD

**Issue by SirHumphreyAppleby** [conformal/cyphertite#102]
Over recent days I have been experiencing some issues with Cyphertite data throughput. Could we please have an update on this issue, and if there is any ETA for a resolution?

Backups are taking longer than they did previously, and it appears Cyphertite is stalling for long periods when transferring new data. Existing files are being processed quickly, suggesting the deduplication process is working. This issue is also occurring when restoring backups, with file restores pausing for long periods.

Everything else appears to be working normally (except for accessing Cyphertite accounts on the Website, which hasn't worked for several days). I can browse backups and see files most of the time, although there have been a couple of occasions where the server could not be queried.

**Comment by sy-p** [comment]
Thanks for your feedback.  We have restored access to the Cyphertite accounts on the website and are looking into any stalling issues that you may have been experiencing.  I'll update when I have more information.

**Comment by ghost** [comment]
FYI being capped at 4Mbps for my backups (I have a 12Mbps upload speed - This speedtest was taken with Cyphertite using 4Mpbs http://www.speedtest.net/my-result/4045553422)

**Comment by SirHumphreyAppleby** [comment]
Throughput seems to have improved again over the last few days. I'm still probably two days away from having statistics from the current backup set, but it looks like we're close to 1MBs^-1 average again.

Another couple of enhancement requests... add time and date information to verbose output so we can see how long individual files are taking and when connections are broken/restored. While I'm certain the throughput problem I was seeing was a legitimate issue, either with Cyphertite or elsewhere, more detailed logging would make evaluating the performance less subjective.

Also, please move statistics output to the exit handler, so it's displayed even if the user interrupts the backup.

**Comment by SirHumphreyAppleby** [comment]
No further stalling issues to report, but speed has dropped right back over the past two days. This may be due to external factors however.

Files scanned                188
Total data          739G    (1.8M/sec)
Data read           739G    (1.8M/sec)
Data skipped            0B
Data compressed         259G    (639K/sec)  (99%)
Data exists         481G    (1.2M/sec)  (65%)
Data sent           258G    (635K/sec)
Reduction ratio              65%
Total Time            425498.732s

**Comment by ghost** [comment]
I can't seem to establish a connection?
http://i.imgur.com/Z7tRIha.jpg

**Comment by zeroryouko** [comment]
I am also experiencing this issue.  Running on Linux (Debian 8.0), I have been uploading data for a few days.  This afternoon the cyphertite command line application crashed, saying there was insufficient free space on the account.  When I run it again, it runs through the completed files very quickly, but when it gets to the file where it crashed, it simply hangs there.  Additionally, running on a Verizon FiOS connection, I am getting an average of 3 Mbit/sec, where it should be closer to 50 Mbit/sec according to Speedtest.net.

**Comment by SirHumphreyAppleby** [comment]
I have been seeing issues again over recent days. An e-mail was sent to support yesterday, but no response has been received,

Backups are failing with a message indicating insufficient space, and then something about the account credentials. Deduplication works quickly, new uploads are stalled. My daily backups are failing... stalled, not going anywhere.

Speed has not been as good as it was during initial testing, so I'm being forced to consider other options.

**Comment by SirHumphreyAppleby** [comment]
Is everyone still experiencing problems?

I've not completed a successful backup since the last days of March. I am not happy about that obviously, since that's the whole point of the service, but also the lack of support or acknowledgement of issues. There are numerous issues with the software (such as incorrect return values and compilation issues) have not been addressed, and urgently need to be. Currently we're paying a lot of money (relative to other options), for what is really a product in beta testing. I'm concerned it may never progress any further, and will be shut down due to lack of resources.

**Comment by dimaj** [comment]
Yes, I am too not able to back anything up and my control panel is showing that my space used / space available are both: not applicable.

I've sent an email to customer support yesterday afternoon. I hope that they'll reply Monday / Tuesday and, more importantly, is that my data is still there and accessible.

**Comment by janinelumc** [comment]
I signed up at the end of March 2015 and i have been getting spotty backups.  At first, when I contacted support, their servers were down and they said they were back up.  Then I had no backups again a few days later.  Another email to support was unanswered and I'm not able to backup at all this weekend.  I've only had this a few weeks but noticed the backups are not running on the weekends.  Will have to see if this particular pattern continues but am wondering how reliable their service is or how good is their technical support.

**Comment by sy-p** [comment]
Over the past 1-2 weeks Cyphertite has experienced a number of unplanned outages.  When we discover these we can get the service back up within an hour but with significantly reduced speeds.

On Thursday and Friday of last week, we addressed the hardware issues behind this.  Since then we've been working to address the software issues.  These may take a few more days to solve but currently the service is up and running better than it had been.  

Thanks.

**Comment by SirHumphreyAppleby** [comment]
When you say " currently the service is up and running better than it had been", does that mean we should be able to backup? My current backup is stalled on a 31.8KB file which happened to change a few days ago.

**Comment by zman0900** [comment]
My backup was stalled too.  Last successful daily backup hasn't happened since April 2nd.  I killed the stalled backup but when I restart it all I get is a "failed to connect" error.

**Comment by sy-p** [comment]
So far today I have tested four different backups of different sizes.  Out of those one got hung up completely and I had to cancel and restart again.  On the second try everything worked fine and finished quickly.

We have not yet identified a reason why back-ups are stalling but we are aware that it is occurring.

**Comment by zman0900** [comment]
I found that my stalled backup had started running this evening and is still going fine a few hours later now.  I was also able to complete another backup on a second machine at the same time.  Fingers crossed that it's fixed for good.

**Comment by SirHumphreyAppleby** [comment]
I am in the process of recreating my entire backup. Fortunately, due to deduplication and the fact that the data is still on the server, this won't take very long. I was deleting everything so I could switch to an unpaid account... i.e. I was ready to abandon the service. Fortunately, sy-p restored some faith in the service by posting an update here. Hopefully things are fixed for good, this time.

**Comment by bobanello** [comment]
I have version 2.0.4 for 64-bit installed.  I am attempting to execute "Delete Server Data" to delete some files.  The first file delete ran 32 minutes and only concluded when I shut down my Windows 8.1 workstation.  The second file delete from the server ran for 17 minutes. None of these are large files.  In the meantime, CDT.EXE is using over 50% of my CPU making it difficult to do other work.  Is this normal for a "Delete Server Data" command?

**Comment by SirHumphreyAppleby** [comment]
I've found deleting archives using the Windows GUI to be unreliable. Deleting old archives (which doesn't remove the data, so may not be doing exactly the same thing) via the Linux command prompt is quick in comparison.

**Comment by SirHumphreyAppleby** [comment]
sy-p, can we have another update (and regular updates) on where things are at please?

I note things have been broken for the last day or so again. With downtime and performance issues, after five months, I am only a little over 50% through uploading my initial backup (mostly historic and won't change or grow significantly going forward). I'd expected this process to be completed next month.

**Comment by sy-p** [comment]
We don't have too much to provide as an update.  We've identified another bad drive that'll need to be replaced and are still troubleshooting the software issues.

**Comment by SirHumphreyAppleby** [comment]
We've had a good couple of weeks, and I've seen significant speed improvements to as high as 1.2MBs^-1. Unfortunately, things seem to have broken again. My backups yesterday morning didn't complete, but after restarting they were finished quickly, suggesting the data had been copied, and only the ct file not created. Since then however, as before, deduplication works, but new files aren't being transferred.

**Comment by sy-p** [comment]
I tested some new files today and measured the same speeds that you were getting for your highs.  

You're running on Linux correct?

**Comment by SirHumphreyAppleby** [comment]
That is correct, yes.

**Comment by SirHumphreyAppleby** [comment]
One new symptom which I don't think I had noted previously is that "ct -me" is also stalling when deleting archives. I probably just didn't notice this previously, as I am using Cyphertite primarily for long term backups. My daily backups (which I am testing with currently) are deleted when they are more than 30 days old.

Currently three days without any backups working, restarting regularly to see if they'll continue.

**Comment by bobanello** [comment]
Starting Thursday, May 28, after 20 days of successful operation, my daily backups started failing to connect.  I also cannot browse my backups on the server.  Anybody having a similar experience?

**Comment by SirHumphreyAppleby** [comment]
Same experience here, and it seems to be the same problems reoccurring. I wouldn't mind so much if there was better communication, and signs that something was actually being done to fix the issues. It also seems that things only get fixed if someone notices they are broken (why isn't this actively monitored?), and never on the weekend.

I need to restore some data from backup, but don't expect to be able to do so for another ~40 hours. Unacceptable really, and I am rapidly running out of patience. I really want to like this service and encourage its growth, since it's in my best interest to recruit more users with lower average data use, but currently it's just not stable enough to recommend to anyone.

**Comment by SirHumphreyAppleby** [comment]
Almost working. Restore running (~20Mb), but hit the out of space error continuing a previously failed backup, and it's not going anywhere after restarting.
